Content.Server/Access/Systems/IdCardConsoleSystem.cs

```diff
@@ -210,4 +226,42 @@ private void UpdateStationRecord(EntityUid uid, EntityUid targetId, string newFu
 
         _record.Synchronize(key);
     }
+
+    private void OnMachineDeconstructed(Entity<IdCardConsoleComponent> entity, ref MachineDeconstructedEvent args)
+    {
+        TryDropAndThrowIds(entity.Owner, entity.Comp, null);
+    }
+
+    private void OnDamageChanged(Entity<IdCardConsoleComponent> entity, ref DamageChangedEvent args)
+    {
+        if (TryDropAndThrowIds(entity.Owner, entity.Comp, null))
+            _chat.TrySendInGameICMessage(entity, Loc.GetString("id-console-system-eject-on-damage"), InGameICChatType.Speak, true);
+    }
+
+    #region Public functions
+    /// <summary>
+    ///     Tries to drop any IDs stored in the console, and then tries to throw them away.
+    ///     Returns true if anything was ejected and false otherwise.
+    /// </summary>
+    public bool TryDropAndThrowIds(EntityUid uid, IdCardConsoleComponent? idCardConsoleComp, ItemSlotsComponent? itemSlotsComp)
+    {
+        if (!Resolve(uid, ref idCardConsoleComp, ref itemSlotsComp))
+            return false;
+
+        var didThrow = false;
+
+        foreach (var slot in itemSlotsComp.Slots.Values)
+        {
+            if (slot.Item == null || slot.ContainerSlot == null)
+                continue;
+
+            var item = slot.Item.Value;
+            _container.Remove(item, slot.ContainerSlot);
+            _throwing.TryThrow(item, _random.NextVector2(), baseThrowSpeed: 5f);
+            didThrow = true;
```

Comment:
I don't have the repo up to check at the moment, but is `TryThrow` actually fallible?
That is to ask, could `TryThrow` end up not throwing, but then `didThrow` gets set to true incorrectly?

Reply:
yeah that doesn't make sense! Changed the logic around a little and renamed stuff to be correct